python/interpret-core/interpret/glassbox/_ebm/_utils.py

```diff
@@ -311,41 +288,34 @@ def deduplicate_bins(bins):
     # use the id of the bins to identify feature data that was previously binned
 
     uniques = dict()
-    for feature_idx in range(len(bins)):
-        bin_levels = bins[feature_idx]
+    for bin_levels in bins:
         highest_key = None
         highest_idx = -1
         for level_idx, feature_bins in enumerate(bin_levels):
             if isinstance(feature_bins, dict):
                 key = frozenset(feature_bins.items())
             else:
                 key = tuple(feature_bins)
-            existing = uniques.get(key, None)
-            if existing is None:
-                uniques[key] = feature_bins
+            if key in uniques:
+                bin_levels[level_idx] = uniques[key]
             else:
-                bin_levels[level_idx] = existing
+                uniques[key] = feature_bins
 
             if highest_key != key:
                 highest_key = key
                 highest_idx = level_idx
-        del bin_levels[highest_idx + 1 :]
+        del bin_levels[highest_idx + 1:]
 
 
 def convert_to_intervals(cuts):  # pragma: no cover
     cuts = np.array(cuts, dtype=np.float64)
 
-    if np.isnan(cuts).any():
-        raise Exception("cuts cannot contain nan")
-
-    if np.isinf(cuts).any():
-        raise Exception("cuts cannot contain infinity")
+    if not np.isfinite(cuts).all():
+        raise Exception("cuts must contain only finite numbers")
 
-    smaller = np.insert(cuts, 0, -np.inf)
-    larger = np.append(cuts, np.inf)
-    intervals = list(zip(smaller, larger))
+    intervals = [(-np.inf, cuts[0]), *zip(cuts[:-1], cuts[1:]), (cuts[-1], np.inf)]
```

Comment:
This will fail if cuts is empty.  Sorry for not having a test for this.

Reply:
No problem, it's a `0.` release after all. The clean solution to handle empty cuts is probably an early return:

```python
if cuts.size == 0:
    return [(-np.inf, np.inf)]
```

Is this your expected output for the function? I didn't find any usage of the two methods, so I am not too sure what they are expected to do.